server/src/main/kotlin/org/javacs/kt/config/Config.kt

```diff
@@ -0,0 +1,45 @@
+package org.javacs.kt.config
+
+import com.google.gson.*
+import org.eclipse.lsp4j.InitializeParams
+import org.javacs.kt.LOG
+import org.javacs.kt.storage.Storage
+import java.lang.reflect.Type
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+fun parseServerConfiguration(params: InitializeParams): ServerConfiguration? {
+    val gson = GsonBuilder().registerTypeHierarchyAdapter(Path::class.java, GsonPathConverter()).create()
+
+    var storage: Storage? = null
+
+    if (params.initializationOptions != null) {
+        val options = gson.fromJson(params.initializationOptions as JsonElement, InitializationOptions::class.java)
+
+        if (options.storagePath != null) {
+            if (Files.exists(options.storagePath) && Files.isDirectory(options.storagePath)) {
+                storage = Storage(options.storagePath)
+            }
+        }
+    }
```

Comment:
Instead of explicit `null` checks, I'd go with `?.let` which is a bit more idiomatic. Also `Files.isDirectory` already covers the existence check, so it shouldn't be necessary:

```suggestion
    params.initializationOptions?.let { initializationOptions ->
        val options = gson.fromJson(initializationOptions as JsonElement, InitializationOptions::class.java)

        options.storagePath?.let { storagePath ->
            if (Files.isDirectory(storagePath)) {
                storage = Storage(storagePath)
            }
        }
    }
```